src/main/java/com/aws/greengrass/clientdevices/auth/iot/ConnectivityInfoProvider.java

```diff
@@ -52,6 +52,9 @@ public ConnectivityInfoProvider(DeviceConfiguration deviceConfiguration,
      * @return list of cached connectivity info items
      */
     public List<String> getCachedHostAddresses() {
```

Comment:
is this a safe change to make, since now this call can block while it waits for a cloud response, right?

Reply:
Good call... should test with EMQx to make sure we're not completely breaking things (if it breaks, that's an EMQx bug).

Reply:
What if there is no IPD, does this work correctly when there is no CIS info at all?

Reply:
It will just get an empty response

Reply:
This shouldn't block cert gen. That process is all async anyway. But we should test it (if it's not, that's still a bug)